### PR TITLE
Try to clarify how ignore-plugin works

### PR DESCRIPTION
**Pull request by DullReferenceException**
The way `ignore-plugin` works wasn't very intuitive to me, so I'd like to try to clarify exactly what the regexes mean in case anyone else is similarly confused.

### CODE REVIEW

File: src/content/plugins/ignore-plugin.md [hunk 1]
```diff
@@ -18,8 +19,28 @@ The following examples demonstrate a few ways this plugin can be used.
 
 ## Ignore Moment Locales
 
-As of [moment](https://momentjs.com/) 2.18, all locales are bundled together with the core library (see [this GitHub issue](https://github.com/moment/moment/issues/2373)). You can use the `IgnorePlugin` to stop any locale being bundled with moment:
+As of [moment](https://momentjs.com/) 2.18, all locales are bundled together with the core library (see [this GitHub issue](https://github.com/moment/moment/issues/2373)). 
+
+Contrary to what you may guess, The `requestRegExp` parameter is not tested against the resolved file names or absolute module
```
Comment (EugeneHlushko) [inline src/content/plugins/ignore-plugin.md]: these line breaks within a paragraph should be removed imo

Comment (EugeneHlushko) [reply src/content/plugins/ignore-plugin.md]: Dont jump into assumption on what the reader might guess or not, this needs rephrasing

File: src/content/plugins/ignore-plugin.md [hunk 2]
```diff
@@ -18,8 +19,28 @@ The following examples demonstrate a few ways this plugin can be used.
...
+names being imported or required, but rather against the _string_ passed to `require` or `import` _within the source code
+where the import is taking place_. For example, if you're trying to exclude `node_modules/moment/locale/*.js`, this won't
+work:
+
+```js
+new webpack.IgnorePlugin(/moment\/locale\//)
+```
+
+Rather, you must know how `moment` itself is importing those files. Inside of the `moment` module, it has this code:
```
Comment (EugeneHlushko) [inline src/content/plugins/ignore-plugin.md]: Because of the way how `moment` itself...

File: src/content/plugins/ignore-plugin.md (hunk 1 above)
Comment (DullReferenceException) [reply src/content/plugins/ignore-plugin.md]: @EugeneHlushko: removed line breaks and did some rewording.

File: src/content/plugins/ignore-plugin.md [hunk 3]
```diff
@@ -19,8 +20,24 @@ The following examples demonstrate a few ways this plugin can be used.
 
 ## Ignore Moment Locales
 
-As of [moment](https://momentjs.com/) 2.18, all locales are bundled together with the core library (see [this GitHub issue](https://github.com/moment/moment/issues/2373)). You can use the `IgnorePlugin` to stop any locale being bundled with moment:
+As of [moment](https://momentjs.com/) 2.18, all locales are bundled together with the core library (see [this GitHub issue](https://github.com/moment/moment/issues/2373)). 
+
+The `requestRegExp` parameter passed to `IgnorePlugin` is not tested against the resolved file names or absolute module names being imported or required, but rather against the _string_ passed to `require` or `import` _within the source code where the import is taking place_. For example, if you're trying to exclude `node_modules/moment/locale/*.js`, this won't work:
+
+```js
```
Comment (EugeneHlushko) [inline src/content/plugins/ignore-plugin.md]: Maybe make this a ```diff and put a minus before non working solution to make it red, we had similar complaints in the past from doc users

**Comment by montogeek** [comment]
Thanks!

**Pull request merged**